docs/error_knowledge_base.md

```diff
@@ -418,14 +419,77 @@ See also: [Submitting a Package](how_to_add_packages.md#submitting-a-package).
 The recipe missess [short_paths](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#short-paths) attribute.
 It may happen due to the very long paths within source, build or package directories during the package creating.
 Consider adding the following code:
+
 ```python
 class SomeRecipe(ConanFile):
     ...
 
     short_paths = True
 ```
 
-See also: (Maximum Path Length Limitation)[https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd].
+See also: [Maximum Path Length Limitation](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd).
+
+#### **<a name="KB-H068">#KB-H068</a>: "TEST_TYPE MANAGEMENT"**
+
+In Conan 2.0, see [migration guide](https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#changes-in-the-test-package-recipe),
+the `test_package/conanfile.py` needs to declare the requirement being tested explicitly. To be prepared you
+have to set the attribute `test_type="explicit"` (this will be ignored in 2.0) to make Conan activate the explicit
+mode, then declaring the requirement using the `self.tested_reference_str` that contains the reference being tested.
+
+#### **<a name="KB-H069">#KB-H069</a>: "TEST PACKAGE - NO DEFAULT OPTIONS"**
+
+This is to ensure the exact package that is built and uploaded is tested against. When `options` of `default_options` are modified in a
+`test_package` it can possibly result in the graph being modified. The objective is to enforce quality of the packages and to avoid confusing
+"missing packages" errors.
+
+#### **<a name="KB-H070">#KB-H070</a>: "MANDATORY SETTINGS"**
+
+> :information_source: This rule was put in place as it was deemed safe for evaluation however there is room for improvement
+
+ConanCenter operates is profiles with a predefined list of settings, all of these much be present in the recipe to ensure `package_id`s are
+computed correctly. Some libraries, for instance header-only, do not require all the settings and those should be deleted from the `package_id`.
+This approach ensure consistency and reduces the learning curve.
+
+Recipes should include:
+
+```python
+class SomeRecipe(ConanFile):
+    ...
+
+    settings = "os", "compiler", "build_type", "arch"
+```
+
+- For header-only recipes ([example](https://github.com/conan-io/conan-center-index/blob/3a773e2d69ada3bd931252c43a48daf636ddfe87/recipes/eigen/all/conanfile.py#L35-L36)):
+
+    ```python
+        def package_id(self):
+            self.info.header_only()
+    ```
+
+- For "build script" recipes ([example](https://github.com/conan-io/conan-center-index/blob/e1b9759bfc2b4625e9ba796499f743ca588d7866/recipes/automake/all/conanfile.py#L47-L50)):
+
+    ```python
+        def package_id(self):
+            del self.info.settings.arch
+            del self.info.settings.compiler
+            del self.info.settings.build_type
+    ```
```

Comment:
Please, explain that a _build-script_ recipe is a package that provides only scripts... no binaries... and I ask you: if it provides only scripts, why it depends on OS?

Reply:
> if it provides only scripts, why it depends on OS?

I am by no means educated on this but AFAIK it has different files for different OS

https://github.com/conan-io/conan-center-index/blob/e1b9759bfc2b4625e9ba796499f743ca588d7866/recipes/automake/all/conanfile.py#L99

https://github.com/conan-io/conan-center-index/pull/3892#discussion_r544619615 was 8,000 PRs ago... maybe @madebr can help use figure out is this is still correct 🤗

Reply:
It was suggested to me to add a `.exe` suffix to the autoconf files to sidestep cci checks failing to accept packages that contain no libraries and no executables.
The addition of the `.exe` is only there for the cci hooks.

Reply:
I think it shouldn't be a common recomendation, ideally, we just add modify hook to allow to capture scripts, or at very least, add exceptions for packages like `automake` that have no executables but just scripts.